src/commands/transaction/accountmetadata.ts

```diff
@@ -0,0 +1,106 @@
+import { command, metadata, option } from 'clime';
+import {
+    AggregateTransaction,
+    Deadline,
+    HashLockTransaction,
+    KeyGenerator,
+    MetadataHttp,
+    MetadataTransactionService,
+    MetadataType,
+    NetworkCurrencyMosaic,
+    PublicAccount,
+    UInt64,
+} from 'nem2-sdk';
+import { AnnounceTransactionsCommand, AnnounceTransactionsOptions } from '../../announce.transactions.command';
+import { MaxFeeResolver } from '../../resolvers/maxFee.resolver';
+import { TargetPublicKeyResolver } from '../../resolvers/publicKey.resolver';
+import { KeyResolver, StringResolver } from '../../resolvers/string.resolver';
+
+export class CommandOptions extends AnnounceTransactionsOptions {
+    @option({
+        flag: 't',
+        description: 'Metadata target public key.',
+    })
+    targetPublicKey: string;
+
+    @option({
+        flag: 'k',
+        description: 'Metadata key scoped to source, target and type.',
+    })
+    key: string;
+
+    @option({
+        flag: 'v',
+        description: 'Metadata value.',
+    })
+    value: string;
+
+    @option({
+        flag: 'd',
+        description: '(Optional) Duration of hash lock.',
+        default: 480,
+    })
+    duration: number;
+
+    @option({
+        flag: 'l',
+        description: 'Lock fee.',
+        default: 10,
+    })
+    lockFee: number;
+}
+
+@command({
+    description: 'Add custom data to an account',
+})
+export default class extends AnnounceTransactionsCommand {
+    constructor() {
+        super();
+    }
+
+    @metadata
+    async execute(options: CommandOptions) {
+        const profile = this.getProfile(options);
+        const account = profile.decrypt(options);
+        const targetAccount = PublicAccount.createFromPublicKey(new TargetPublicKeyResolver().resolve(options), account.networkType);
+        const key = KeyGenerator.generateUInt64Key(new KeyResolver().resolve(options));
```

Comment:
The key should not be transformed using ``KeyGenerator.generateUInt64Key(key)`` by default. The command should accept as a parameter a UInt64 string instead. The user will be able to encode the string value as a key using this other command: https://github.com/nemtech/nem2-cli/pull/178/files#diff-9fe1ecd7764fb484405286123c7e83f1 (applies to the other commands)